src/OpenApi/gen/Helpers/AssemblyTypeSymbolsVisitor.cs

```diff
@@ -70,13 +70,16 @@ public override void VisitNamedType(INamedTypeSymbol type)
         foreach (var method in methods)
         {
             _cancellationToken.ThrowIfCancellationRequested();
-            if (IsDeclaredInAssembly(method) && _exportedMethods.Add(method))
+            if (IsAccessibleType(method) && _exportedMethods.Add(method))
             {
                 method.Accept(this);
             }
         }
     }
 
-    private bool IsDeclaredInAssembly(ISymbol symbol) =>
-        SymbolEqualityComparer.Default.Equals(symbol.ContainingAssembly, assemblySymbol);
+    private bool IsAccessibleType(ISymbol symbol) =>
+        SymbolEqualityComparer.Default.Equals(symbol.ContainingAssembly, assemblySymbol)
+            && (symbol.DeclaredAccessibility == Accessibility.Public ||
```

Comment:
Is it simpler to check for `!= Accessibility.Private`? Or are there more accessibility states I'm forgetting?

Reply:
The way the [Accessibility enum](https://sourceroslyn.io/#Microsoft.CodeAnalysis/Symbols/Accessibility.cs,96ec9a52e8b1305b) is modeled is kinda weird. It does it based on the modifiers in the class so `protected class Foo` will have `Accessibility.Protected` even though it is private.